modules/objdetect/src/qrcode.cpp

```diff
@@ -1001,212 +1069,1332 @@ void QRDecode::init(const Mat &src, const vector<Point2f> &points)
+            Point p1, p2;
+            int index_p1_in_vertices = 0, index_p2_in_vertices = 0;
+            for (int k = 4; k > 0; k--)
+            {
+                if(((int)vertices_dist_pair[0].first == k % 4) && ((int)vertices_dist_pair[1].first == (k - 1) % 4))
+                {
+                    index_p1_in_vertices = vertices_dist_pair[0].first;
+                    index_p2_in_vertices = vertices_dist_pair[1].first;
+                }
+                else if(((int)vertices_dist_pair[1].first == k % 4) && ((int)vertices_dist_pair[0].first == (k - 1) % 4))
+                {
+                    index_p1_in_vertices = vertices_dist_pair[1].first;
+                    index_p2_in_vertices = vertices_dist_pair[0].first;
+                }
+            }
+            if (index_p1_in_vertices == index_p2_in_vertices) return false;
+            
+            p1 = vertices[index_p1_in_vertices];
+            p2 = vertices[index_p2_in_vertices];
+
+            size_t index_p1_in_contour = 0, index_p2_in_contour = 0;    
+            vector<Point> add_points = patterns_contours[patterns_indexes[j]];
+
+            for(size_t k = 0; k < add_points.size(); k++)
+            {
+                if (add_points[k] == p1)
+                {
+                    index_p1_in_contour = k;
+                }
+                if (add_points[k] == p2)
+                {
+                    index_p2_in_contour = k;
+                }
+            }
+
+            if (index_p1_in_contour > index_p2_in_contour)
+            {
+                for (size_t k = index_p1_in_contour; k < add_points.size(); k++)
+                {
+                    points.push_back(add_points[k]);
+                }
+                for (size_t k = 0; k <= index_p2_in_contour; k++)
+                {
+                    points.push_back(add_points[k]);
+                }
+            }
+            else if (index_p1_in_contour < index_p2_in_contour)
+            {
+                for (size_t k = index_p1_in_contour; k <= index_p2_in_contour; k++)
+                {
+                    points.push_back(add_points[k]);
+                }
+            }
+            else { return false; }
+            if (abs(p1.x - p2.x) > abs(p1.y - p2.y))
+            {
+                sort(points.begin(), points.end(), sortPointsByX());
+            }
+            else
+            {
+                sort(points.begin(), points.end(), sortPointsByY());
+            }
+
+            temp_patterns_add_points.push_back(std::pair<int, vector<Point> >(idx_curved_side,points));
+
+        }
+    }
+    
+    return true;
+}
+
+bool QRDecode::computePatternsAddingPoints(std::map<int, vector<Point> > &patterns_add_points)
+{
+    vector<std::pair<int, vector<Point> > > temp_patterns_add_points;
+    if(!findTempPatternsAddingPoints(temp_patterns_add_points)) { return false; }
+    for(size_t i = 0; i < temp_patterns_add_points.size(); i++)
+    {
+        int num_points_in_pattern = 3;
```

Comment:
```suggestion
        const int num_points_in_pattern = 3;
```
make all constant variables `const`